lib/utils/googleAuth.ts

```diff
@@ -133,12 +133,16 @@ export const hasReadOnlyPermissionOnly = (
 };
 
 export const getRedirect = (
-  user?: HackneyGoogleUserWithPermissions
+  user?: HackneyGoogleUserWithPermissions,
+  checkForReadOnlyPermissions?: boolean
```

Comment:
Don't we always want to ensure that users with read-only permissions can do only what's allowed?

Reply:
Nevermind, I see what the issue is: the `getRedirect` is used by both the pages that the read-only user is allowed on, and the on the pages that they're not allowed on.

So this function needs a way to distinguish between whenever they should be allowed on and when not. Perhaps, renaming this variable would be useful? Like:

```ts
writePermissionsRequired?: boolean
```

Reply:
Yes, I think that's a bit more suitable name for it. Updated in https://github.com/LBHackney-IT/lbh-housing-register/pull/399/commits/794a0841e84e41b379f783a633b71f6a7d501b61